File: python/design-patterns/proxy/naive/document_naive.py
class Document:
    def __init__(self, files):
        # ❌ every image decodes NOW, even ones never viewed
        self.images = [RealImage(f) for f in files]   # ~24MB decode EACH
    def show(self, i): return self.images[i].display()

doc = Document(["p1.raw", "p2.raw"])  # + 48 more
# All decoded up front — slow open, huge memory.

File: go/design-patterns/proxy/naive/document_naive.go
type Document struct{ images []*RealImage }

func NewDocument(files []string) *Document {
    d := &Document{}
    // ❌ every image decodes NOW, even ones never viewed
    for _, f := range files {
        d.images = append(d.images, NewRealImage(f)) // ~24MB EACH
    }
    return d
}
// NewDocument(fiftyFiles) -> 50 decodes up front.